### PR TITLE
Arch: adapt materials to new Materials path scheme

### LINKED ISSUE #13821: Materials tool crashes on Arch workbench
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Steps to reproduce

1. Create a new document
2. Switch to the Arch workbench
3. Click on the <kbd>Material Tools</kbd> ([Arch_MaterialTools](https://wiki.freecad.org/Arch_CompSetMaterial)) icon
4. Material tools don't load, the user is presented with this error

```python-traceback
12:00:06  pyException: Traceback (most recent call last):
  File "/usr/share/freecad-daily/Mod/Arch/ArchMaterial.py", line 481, in setEdit
    self.taskd = _ArchMaterialTaskPanel(vobj.Object)
  File "/usr/share/freecad-daily/Mod/Arch/ArchMaterial.py", line 556, in __init__
    self.fillMaterialCombo()
  File "/usr/share/freecad-daily/Mod/Arch/ArchMaterial.py", line 699, in fillMaterialCombo
    for f in os.listdir(p):
<class 'FileNotFoundError'>: [Errno 2] No such file or directory: '/usr/share/freecad-daily//Mod/Material/StandardMaterial'
```

I can reproduce this both with the PPA and snap package builds, as well as with a local built. So it seems it's not simply a packaging issue. It can also be reproduced with the BIM workbench.

It looks like the issue surfaces in this part of the code:
https://github.com/FreeCAD/FreeCAD/blob/9bd19ff43a1c984d9d241f85c31c3d30be876d06/src/Mod/Arch/ArchMaterial.py#L689-L705

I believe the code is looking for the standard materials folder here:
`$RESOURCE_DIR/Mod/Material/StandardMaterial/`

But the new Material workbench revamp has changed that location to:
`$RESOURCE_DIR/Mod/Material/Resources/Materials/Standard/`

### Full version info

```shell
OS: Ubuntu 22.04.4 LTS (ubuntu:GNOME/ubuntu)
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.37096 (Git)
Build type: Release
Branch: main
Hash: af2b8b661f711f8512d63345e4ffe321e292ddbb
Python 3.10.12, Qt 5.15.3, Coin 4.0.0, Vtk 9.1.0, OCC 7.6.3
Locale: C/Default (C)
Installed mods: 
  * OpenTheme 2024.4.20
  * BIM 2021.12.0
  * parts_library
```

### Subproject(s) affected?

Arch

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by furgo16**
Fixes #13821 by adapting the old material paths to the new scheme:

1. Update the materials filesystem path when reading them
2. Materials can now be in subfolders: traverse directories to read all materials

I'm aware of https://github.com/FreeCAD/FreeCAD/issues/13821#issuecomment-2094225362. I've addressed this as a quicker way to get back the Materials function before https://github.com/FreeCAD/FreeCAD/pull/13783 gets merged, which might take longer. The intention is certainly not to step on anyone's toes, so if this is already being worked on, or can be fixed in a better way, feel free to close this PR.

### CODE REVIEW

File: src/Mod/Material/importFCMat.py [hunk 1]
```diff
@@ -97,9 +97,9 @@ def decode(name):
 # as we had and we might will have problems again and again
 # https://github.com/berndhahnebach/FreeCAD_bhb/commits/materialdev
 
-def read(filename):
+def read(filename, library):
```
Comment (davesrocketshop) [inline src/Mod/Material/importFCMat.py:100]: Unfortunately this code will only read materials from the standard library, not user materials, etc. Specifying the library is not required when the full path is required, and when using the old system is not recommended.

This change will also break the many add on workbenches that call this function.

**Review by davesrocketshop — CHANGES_REQUESTED** [review]
It should only be necessary to change the path. The other changes will break other workbenches.

File: src/Mod/Material/importFCMat.py (hunk 1 above)
Comment (furgo16) [reply src/Mod/Material/importFCMat.py:100]: Thanks for the review. Reverted changes, now calling this function from ArchMaterial.py with absolute material path.

File: src/Mod/Arch/ArchMaterial.py [hunk 2]
```diff
@@ -652,7 +652,7 @@ def chooseMat(self, card):
         "sets self.material from a card"
         if card in self.cards:
             import importFCMat
-            self.material = importFCMat.read(self.cards[card])
+            self.material = importFCMat.read(self.cards[card][0], self.cards[card][1])
```
Comment (furgo16) [inline src/Mod/Arch/ArchMaterial.py:655]:
```suggestion
            self.material = importFCMat.read(self.cards[card])
```
Missed to revert this change on the last commit

**Comment by furgo16** [comment]
@davesrocketshop thanks again for doing the review. I believe the new changes address your review comments. Let me know if that looks ok.

**Comment by davesrocketshop** [comment]
> @davesrocketshop thanks again for doing the review. I believe the new changes address your review comments. Let me know if that looks ok.

It does. You might also want to look at _materialtools.cardutils.get_material_resources()_ that builds the path for you including user materials, custom folders, and any added by add on workbenches. The need goes away though as the module transitions to the new materials system so that may not be necessary immediately.

**Comment by furgo16** [comment]
Thanks. I'll look into `get_material_resources()` for the next iteration. I had wondered if there was a dedicated function, but I had not found it. I'll look at the code examples I see in the `cardutils.py` comments in more detail, then.
